packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -91,6 +91,7 @@ class SemanticsData extends Diagnosticable {
     @required this.value,
     @required this.decreasedValue,
     @required this.hint,
+    @required this.routeName,
```

Comment:
Should this also be made available to the CustomPainterSemantics? @yjbanov

Reply:
`CustomPaint` takes a `Widget child`, which can set it, or one of its ancestors can. `CustomPainter`s do not add any more widget subtrees otherwise. So my guess is, no? I can be convinced otherwise though.

Reply:
I don't have a strong opinion either and this can easily be added later. Just one more thought pro adding it (at some point):

Flutter is implemented using Custom Views on Android and iOS. It's possible that in the distant future somebody wants to implement a Flutter-like framework on top of Flutter (just like we do on top of Android/iOS) using Flutter's CustomPaint concept. They'll need a way to announce route names.

Reply:
That's not a reasonable way to extend Flutter, I wouldn't worry about it. If you want to extend Flutter, you just do so right at the same layer we do.